### PR TITLE
Article about Food Discovery demo

### PR DESCRIPTION
**Pull request by kacperlukawski**

### CODE REVIEW

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+Since the [Recommendation API](https://qdrant.tech/documentation/concepts/search/#recommendation-api) requires at least one positive example, we can use it only when 
+the user has liked at least one dish. We could theoretically use the same trick as above and negate the disliked dishes, but it would be a bit weird, as Qdrant has
+that feature already built-in, and we can call it just once. It's always better to perform the search server-side. Thus, in this case we just call the Qdrant server
+with a list of positive and negative examples, so it can find some points which are close to the positive examples and far from the negative ones.
+
+TODO: code sample
+
+### Location-based search
+
+Last but not least. Location plays an important role in the food discovery process. You are definitely looking for something you can find nearby, not on the other
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:171]:
```suggestion
Last but not least, location plays an important role in the food discovery process. You are definitely looking for something you can find nearby, not on the other
```

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                ),
+                radius=location.radius_km * 1000,
+            ),
+        )
+    ]
+)
+```
+
+Such a filter needs [a payload index](https://qdrant.tech/documentation/concepts/indexing/#payload-index) to work efficiently, and it was created on a collection
+we used to create the snapshot. So if you import it, the index will be already there.
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:197]: Cant start the sentences with So, And etc.

**Comment by davidmyriel** [comment]
I recommend compressing the paragraphs for easy reading. If you have just one thought, make it one sentence.
Otherwise, the whole article makes good sense and the procedure is well highlighted.

**Review by davidmyriel — CHANGES_REQUESTED** [review]

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+### Textual search
+
+Since the demo suffers from the cold start problem, we implemented a textual search mode that is useful to start exploring the data. You can type in any text query
+by clicking a search icon in the top right corner. The demo will use the CLIP model to encode the query into a vector and then search for the nearest neighbors
+in the vector space. 
+
+TODO: add gif showing how it works in UI
+
+Under the hood this is [a group search query to Qdrant](https://github.com/qdrant/demo-food-discovery/blob/6b49e11cfbd6412637d527cdd62fe9b9f74ac699/backend/discovery.py#L44).
```
[the remark this answers is not in the document]
Comment (kacperlukawski) [reply qdrant-landing/content/articles/food-discovery-demo.md]: I tend to use it quite often, but I thought it was a popular term:
https://en.wiktionary.org/wiki/under_the_hood

**Comment by kacperlukawski** [comment]
@davidmyriel Thanks for your review. I'd appreciate another one, as I modified the article a bit. There are some sketches to visualize concepts that must be replaced with better graphics, and we need to deploy the demo somewhere to expose it publicly. Other than that, I don't think there is much to add.

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
+---
+title: Food Discovery Demo
+short_description: Feeling hungry? Multimodal semantic search makes it easy to find the perfect meal.
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:3]:
```suggestion
short_description: Feeling hungry? Find the perfect meal with Qdrant's multimodal semantic search.
```

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+The demo will be available at `http://localhost:8000`, but you won't be able to search anything until you import the snapshot into your Qdrant instance.
+If you don't want to bother with hosting a local one, you can use the [Qdrant Cloud](https://cloud.qdrant.io/) cluster. 4 GB RAM is enough to load all the 2 million
+entries.
+
+## Fork and reuse
+
+The full source code of this demonstration is open-sourced and available for you to fork, adapt, and build upon. Whether you’re looking to understand the mechanics 
+of semantic search or to have a foundation to build a larger project, this demo can serve as a starting point. Check out the [Food Discovery Demo repository
+](https://github.com/qdrant/demo-food-discovery/) to get started. If you have any questions, feel free to reach out to us [on our Discord community](https://qdrant.to/discord).
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:233]:
```suggestion
](https://github.com/qdrant/demo-food-discovery/) to get started. If you have any questions, feel free to reach out  [through Discord](https://qdrant.to/discord).
```

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+[README](https://github.com/qdrant/demo-food-discovery/blob/main/README.md) describes all the steps more in detail, but here is a quick start:
+
+```bash
+git clone git@github.com:qdrant/demo-food-discovery.git
+cd demo-food-discovery
+# Create .env file based on .env.example
+docker-compose up -d
+```
+
+The demo will be available at `http://localhost:8000`, but you won't be able to search anything until you import the snapshot into your Qdrant instance.
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:225]:
```suggestion
The demo will be available at `http://localhost:8000`, but you won't be able to search for anything until you import the snapshot into your Qdrant instance.
```

Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:225]: import the snapshot into your Qdrant instance - should link to instructions for how to do this

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+    limit=search_query.limit,
+)
+```
+
+From the user perspective nothing changes comparing to the previous case.
+
+### Location-based search
+
+Last but not least, location plays an important role in the food discovery process. You are definitely looking for something you can find nearby, not on the other
+side of the globe. Therefore, all the search modes accepts your current location as a filtering criteria, and you can enable it by clicking on “Find near me” icon
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:186]:
```suggestion
side of the globe. Therefore, your current location can be toggled as a filtering condition. You can enable it by clicking on “Find near me” icon
```

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+we want to be able to provide only negative examples. This is because we want to be able to say “I don’t like this dish” without having to like anything first.
+To achieve this, we use a trick. We negate the vectors of the disliked dishes and use their mean as a query. This way, the disliked dishes will be pushed away 
+from the search results. **This works because the cosine distance is based on the angle between two vectors, and the angle between a vector and its negation is 180 degrees.**
+
+![CLIP model](/articles_data/food-discovery-demo/negated-vector.png)
+
+Food Discovery Demo [implements that trick](https://github.com/qdrant/demo-food-discovery/blob/6b49e11cfbd6412637d527cdd62fe9b9f74ac699/backend/discovery.py#L122)
+by calling Qdrant twice. First of all we use the [Scroll API](https://qdrant.tech/documentation/concepts/points/#scroll-points) to find the disliked items, 
+and then calculate a negated mean of all their vectors. That allows using the [Search Groups API](https://qdrant.tech/documentation/concepts/search/#search-groups)
+to find the nearest neighbors of the negated mean vector and we are there!
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:133]:
```suggestion
to find the nearest neighbors of the negated mean vector.
```

File: qdrant-landing/content/articles/food-discovery-demo.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+Qdrant [Recommendation API](https://qdrant.tech/documentation/concepts/search/#recommendation-api) needs at least one positive example to work. However, in our demo
+we want to be able to provide only negative examples. This is because we want to be able to say “I don’t like this dish” without having to like anything first.
+To achieve this, we use a trick. We negate the vectors of the disliked dishes and use their mean as a query. This way, the disliked dishes will be pushed away 
+from the search results. **This works because the cosine distance is based on the angle between two vectors, and the angle between a vector and its negation is 180 degrees.**
+
+![CLIP model](/articles_data/food-discovery-demo/negated-vector.png)
+
+Food Discovery Demo [implements that trick](https://github.com/qdrant/demo-food-discovery/blob/6b49e11cfbd6412637d527cdd62fe9b9f74ac699/backend/discovery.py#L122)
+by calling Qdrant twice. First of all we use the [Scroll API](https://qdrant.tech/documentation/concepts/points/#scroll-points) to find the disliked items,
```
Comment (davidmyriel) [inline qdrant-landing/content/articles/food-discovery-demo.md:131]:
```suggestion
by calling Qdrant twice. Initially, we use the [Scroll API](https://qdrant.tech/documentation/concepts/points/#scroll-points) to find disliked items, 
```

File: qdrant-landing/content/articles/food-discovery-demo.md (hunk 4 above)
Comment (davidmyriel) [reply qdrant-landing/content/articles/food-discovery-demo.md:3]: Give me a few more rounds. I really wanna do this well. The demo could be a big hit.

File: qdrant-landing/content/articles/food-discovery-demo.md (hunk 6 above)
Comment (kacperlukawski) [reply qdrant-landing/content/articles/food-discovery-demo.md]: It has been linked to the documentation.

**Comment by kacperlukawski** [comment]
@davidmyriel I included the suggested changes and included the updated diagrams created by Juan.

**Review by davidmyriel — APPROVED** [review]

File: qdrant-landing/static/articles_data/food-discovery-demo/preview/social_preview.jpg
Comment (generall) [inline qdrant-landing/static/articles_data/food-discovery-demo/preview/social_preview.jpg:1]: Please use template from figma for preview

Comment (kacperlukawski) [reply qdrant-landing/static/articles_data/food-discovery-demo/preview/social_preview.jpg:1]: Done, the social preview image is changed

**Pull request merged**